test-data/unit/check-functools.test

```diff
@@ -0,0 +1,250 @@
+[case testTotalOrderingEqLt]
+from functools import total_ordering
+
+@total_ordering
+class Ord:
+    def __eq__(self, other: object) -> bool:
+        return False
+
+    def __lt__(self, other: "Ord") -> bool:
+        return False
+
+Ord() < Ord()
+Ord() <= Ord()
+Ord() == Ord()
+Ord() > Ord()
+Ord() >= Ord()
+
+Ord() < 1  # E: Unsupported operand types for < ("Ord" and "int")
+Ord() <= 1  # E: Unsupported operand types for <= ("Ord" and "int")
+Ord() == 1
+Ord() > 1  # E: Unsupported operand types for > ("Ord" and "int")
+Ord() >= 1  # E: Unsupported operand types for >= ("Ord" and "int")
+[builtins fixtures/ops.pyi]
+[builtins fixtures/dict.pyi]
+
+[case testTotalOrderingEqLe]
```

Comment:
The remaining tests are a bit verbose. Maybe simplify some of them and only test cases that are expected to be materially different from other test cases? Another option is to vary the test cases in some aspects. Here are some ideas (some of these could be separate tests as well):

* Omit the function type annotation in some test case (fall back to `Any` types).
* Use a different argument type in some test case.
* Use an unusual return type in some test case, such as `str`.
* Test non-callable `__lt__`.
* Test method defined as a variable (`__lt__ = lambda: ...`, for example).